tests/gui/components/update-notifier.spec.js

```diff
@@ -3,146 +3,112 @@
-            const result = UpdateNotifierService.shouldCheckForUpdates({
-              ignoreSleepUpdateCheck: this.ignoreSleepUpdateCheck
-            });
-
-            m.chai.expect(result).to.be.true;
+        it('should return true if ignoreSleepUpdateCheck is false', function() {
+          const result = UpdateNotifierService.shouldCheckForUpdates({
+            ignoreSleepUpdateCheck: false,
+            lastSleptUpdateNotify: undefined
           });
 
+          m.chai.expect(result).to.be.true;
         });
 
-        describe('given the `sleepUpdateCheck` is enabled', function() {
-
-          beforeEach(function() {
-            settings.set('sleepUpdateCheck', true);
+        it('should return true if ignoreSleepUpdateCheck is true', function() {
+          const result = UpdateNotifierService.shouldCheckForUpdates({
+            ignoreSleepUpdateCheck: true,
+            lastSleptUpdateNotify: undefined
           });
 
-          describe('given the `lastUpdateNotify` was never updated', function() {
-
-            beforeEach(function() {
-              settings.set('lastUpdateNotify', undefined);
-            });
+          m.chai.expect(result).to.be.true;
+        });
 
-            it('should return true', function() {
-              const result = UpdateNotifierService.shouldCheckForUpdates({
-                ignoreSleepUpdateCheck: this.ignoreSleepUpdateCheck
-              });
+      });
 
-              m.chai.expect(result).to.be.true;
-            });
+      describe('given the `lastSleptUpdateNotify` was very recently updated', function() {
 
+        it('should return false if ignoreSleepUpdateCheck is false', function() {
+          const result = UpdateNotifierService.shouldCheckForUpdates({
+            ignoreSleepUpdateCheck: false,
+            lastSleptUpdateNotify: Date.now() - 1000
           });
 
-          describe('given the `lastUpdateNotify` was very recently updated', function() {
-
-            beforeEach(function() {
-              settings.set('lastUpdateNotify', Date.now() + 1000);
-            });
-
-            it('should return false', function() {
-              const result = UpdateNotifierService.shouldCheckForUpdates({
-                ignoreSleepUpdateCheck: this.ignoreSleepUpdateCheck
-              });
-
-              m.chai.expect(result).to.be.false;
-            });
+          m.chai.expect(result).to.be.false;
+        });
 
+        it('should return true if ignoreSleepUpdateCheck is true', function() {
+          const result = UpdateNotifierService.shouldCheckForUpdates({
+            ignoreSleepUpdateCheck: true,
+            lastSleptUpdateNotify: Date.now() - 1000
           });
 
-          describe('given the `lastUpdateNotify` was updated long ago', function() {
-
-            beforeEach(function() {
-              const SLEEP_MS = units.daysToMilliseconds(UPDATE_NOTIFIER_SLEEP_DAYS);
-              settings.set('lastUpdateNotify', Date.now() + SLEEP_MS + 1000);
```

Comment:
Indeed, which is terrible :/ . I clearly have major issues reasoning about this component, so lets join forces to make sure I get it right this time (and that the tests really test that).